propagation.go

```diff
@@ -0,0 +1,103 @@
+package opentracing
+
+///////////////////////////////////////////////////////////////////////////////
+// CORE PROPAGATION INTERFACES:
+///////////////////////////////////////////////////////////////////////////////
+
+// PropagationInjector is responsible for injecting Span instances in a manner
+// suitable for propagation via a format-specific "carrier" object. Typically
+// the injection will take place across an RPC boundary, but message queues and
+// other IPC mechanisms are also reasonable places to use a
+// PropagationInjector.
+//
+// See PropagationExtractor and Span.PropagationInjectorForFormat.
+type PropagationInjector interface {
+	InjectSpan(span Span, carrier interface{})
+}
+
+// PropagationExtractor is responsible for extracting Span instances from an
+// format-specific "carrier" object. Typically the extraction will take place
+// on the server side of an RPC boundary, but message queues and other IPC
+// mechanisms are also reasonable places to use a PropagationExtractor.
+//
+// See PropagationInjector and Tracer.PropagationExtractorForFormat.
+type PropagationExtractor interface {
+	ExtractSpan(operationName string, carrier interface{}) (Span, error)
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// BUILTIN PROPAGATION FORMATS:
+///////////////////////////////////////////////////////////////////////////////
+
+// BuiltinPropagationFormat is the shared type for builtin OpenTracing in-band
+// propagation formats.
+type BuiltinPropagationFormat int
+
+const (
+	// PROPAGATION_FORMAT_SPLIT_BINARY encodes the Span in a BinaryCarrier
+	// instance.
+	//
+	// The `carrier` for injection and extraction must be a `*BinaryCarrier`
+	// instance.
+	PROPAGATION_FORMAT_SPLIT_BINARY BuiltinPropagationFormat = iota
```

Comment:
agreed. And does it have to be an enum? Why not just

```go
package encoding

type Binary struct{}
type HTTP struct{}
```

that's what would happen for custom wire types elsewhere.